cobrawap/pipeline/stage04_wave_detection/scripts/optical_flow.py

```diff
@@ -98,60 +104,81 @@ def compute_derivatives(frame, next_frame, kernelX, kernelY,
+                     alpha=alpha,
+                     max_Niter=max_Niter,
+                     convergence_limit=convergence_limit,
+                     kernelHS=kernelHS,
+                     kernelT=kernelT,
+                     kernelX=kernelX,
+                     kernelY=kernelY,
+                     are_phases=are_phases)
+                    for i in tqdm(range(len(frames[:-1])), ascii=True))
+    
+    vector_frames = np.asarray(vector_frames, dtype=complex)
+    vector_frames[:,nan_channels[0],nan_channels[1]] = np.nan + np.nan*1j
 
     frames[:,nan_channels[0],nan_channels[1]] = np.nan
     return vector_frames
 
 
-def interpolate_empty_sites(frames, are_phases=False):
-    if np.isfinite(frames).all():
-        return frames
-    dim_y, dim_x = frames[0].shape
-    grid_y, grid_x = np.meshgrid([-1,0,1],[-1,0,1], indexing='ij')
-
-    for i, frame in enumerate(frames):
-        new_frame = copy(frame)
-        while not np.isfinite(new_frame).all():
-            y, x = np.where(np.bitwise_not(np.isfinite(new_frame)))
+def interpolation_step(frame, grid_y, grid_x, are_phases=False, max_iters=1000):
+    dim_y, dim_x = frame.shape
+    if np.isfinite(frame).all():
+        return frame
+    else:
+        for _ in range(max_iters):
+            new_frame = copy(frame)
+            y, x = np.where(np.bitwise_not(np.isfinite(frame)))
             # loop over nan-sites
-            for xi, yi in zip(x,y):
+            for xi, yi in zip(x, y):
+
                 neighbours = []
                 # collect neighbours of each site
                 for dx, dy in zip(grid_x.flatten(), grid_y.flatten()):
-                    xn = xi+dx
-                    yn = yi+dy
+                    xn = xi + dx
+                    yn = yi + dy
                     if (0 <= xn) & (xn < dim_x) & (0 <= yn) & (yn < dim_y):
-                        neighbours.append(frames[i, yn, xn])
+                        neighbours.append(frame[yn, xn])
                 # average over neihbour values
                 if np.isfinite(neighbours).any():
                     if are_phases:
-                        vectors = np.exp(1j*np.array(neighbours))
-                        new_frame[yi,xi] = np.angle(np.nansum(vectors))
+                        vectors = np.exp(1j * np.array(neighbours))
+                        new_frame[yi, xi] = np.angle(np.nansum(vectors))
                     else:
-                        new_frame[yi,xi] = np.nansum(neighbours)
-            frames[i] = new_frame
-    return frames
+                        new_frame[yi, xi] = np.nansum(neighbours)
+            frame = new_frame
+            if np.isfinite(frame).all():
+                break
+        return frame
+
+
+def interpolate_empty_sites(frames, are_phases=False, max_iters=1000, n_jobs=1):
+    if np.isfinite(frames).all():
+        return frames
+    else:
+        grid_y, grid_x = np.meshgrid([-1, 0, 1], [-1, 0, 1], indexing='ij')
+        print('Frames interpolation')
```

Comment:
remove print statement. instead you could set this as a title for tqdm

Reply:
Ok, we will add a title to the tqdm progress bar.